x-pack/plugins/gis/public/components/layer_panel/settings_panel/settings_panel.js

```diff
@@ -4,7 +4,7 @@
  * you may not use this file except in compliance with the Elastic License.
```

Comment:
ugh, I absolute hate that function syntax for a react component. so cluttered.

Reply:
But it is more performant since there are not state checks or life cycle methods

Reply:
interesting. but if we want to avoid React internally to do this whole life-cycle loop, we would also need to re-implement how we include the component.

I just got this from:
https://medium.com/missive-app/45-faster-react-functional-components-now-3509a668e69f